primitives/rpc/src/lib.rs

```diff
@@ -177,8 +177,26 @@ sp_api::decl_runtime_apis! {
 		/// Return the elasticity multiplier.
 		fn elasticity() -> Option<Permill>;
 	}
+
+	#[api_version(2)]
+	pub trait ConvertTransactionRuntimeApi {
+		fn convert_transaction(transaction: ethereum::TransactionV2) -> <Block as BlockT>::Extrinsic;
+		#[changed_in(2)]
+		fn convert_transaction(transaction: ethereum::TransactionV0) -> <Block as BlockT>::Extrinsic;
+	}
 }
 
 pub trait ConvertTransaction<E> {
 	fn convert_transaction(&self, transaction: ethereum::TransactionV2) -> E;
 }
+
+// `NoTransactionConverter` is a non-instantiable type (an enum with no variants),
+// so we are guaranteed at compile time that `NoTransactionConverter` can never be instantiated.
+pub enum NoTransactionConverter {}
```

Comment:
Is this actually used?

Reply:
If you have this type, then why do you still have an `Option<_>` in `Eth::new`?

Reply:
Oh, I get it. I use this only as a type parameter. and use `None` as the value. That's why it's not instantiable.